src/librustc_lint/builtin.rs

```diff
@@ -227,6 +227,14 @@ impl UnsafeCode {
 }
 
 impl<'a, 'tcx> LateLintPass<'a, 'tcx> for UnsafeCode {
+    fn check_attribute(&mut self, cx: &LateContext, attr: &ast::Attribute) {
+        if attr.check_name("allow_internal_unsafe") {
+            self.report_unsafe(cx, attr.span, "`allow_internal_unsafe` allows defining \
```

Comment:
You can try running `cx.span_lint(UNSAFE_CODE, attr.span, "message");` directly, instead of going through `report_unsafe`. Maybe the attribute already overlaps with the "allows_unsafe" check in the `report_unsafe` method.

Reply:
I changed to `cx.span_lint(UNSAFE_CODE, attr.span, "description");`, but stdout and stderr were nothing. 
```
stdout:
------------------------------------------

------------------------------------------
stderr:
------------------------------------------

------------------------------------------

thread '[ui] ui\lint\lint-forbid-allow-internal-unsafe.rs' panicked at 'explicit panic', src\tools\compiletest\src\runtest.rs:3245:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.


failures:
    [ui] ui\lint\lint-forbid-allow-internal-unsafe.rs

test result: FAILED. 5265 passed; 1 failed; 35 ignored; 0 measured; 0 filtered out

thread 'main' panicked at 'Some tests failed', src\tools\compiletest\src\main.rs:495:22
```

Reply:
backtrace is here
```
stack backtrace:
   0: std::sys::windows::backtrace::set_frames
             at /rustc/a01e4761a1507939430d4044a5f0a35fdb2b146c\/src\libstd\sys\windows\backtrace\mod.rs:104
   1: std::sys::windows::backtrace::set_frames
             at /rustc/a01e4761a1507939430d4044a5f0a35fdb2b146c\/src\libstd\sys\windows\backtrace\mod.rs:104
   2: std::sys::windows::backtrace::set_frames
             at /rustc/a01e4761a1507939430d4044a5f0a35fdb2b146c\/src\libstd\sys\windows\backtrace\mod.rs:104
   3: std::sys_common::backtrace::print
             at /rustc/a01e4761a1507939430d4044a5f0a35fdb2b146c\/src\libstd\sys_common\backtrace.rs:59
   4: std::sys_common::backtrace::print
             at /rustc/a01e4761a1507939430d4044a5f0a35fdb2b146c\/src\libstd\sys_common\backtrace.rs:59
   5: std::panicking::default_hook
             at /rustc/a01e4761a1507939430d4044a5f0a35fdb2b146c\/src\libstd\panicking.rs:227
   6: std::panicking::rust_panic_with_hook
             at /rustc/a01e4761a1507939430d4044a5f0a35fdb2b146c\/src\libstd\panicking.rs:491
   7: std::panicking::begin_panic
   8: <core::marker::PhantomData<T> as serde::de::DeserializeSeed<'de>>::deserialize
   9: <serde::de::impls::<impl serde::de::Deserialize<'de> for alloc::vec::Vec<T>>::deserialize::VecVisitor<T> as serde::de::Visitor<'de>>::visit_seq
  10: std::rt::lang_start_internal::{{closure}}
             at /rustc/a01e4761a1507939430d4044a5f0a35fdb2b146c\/src\libstd\rt.rs:59
  11: std::rt::lang_start_internal::{{closure}}
             at /rustc/a01e4761a1507939430d4044a5f0a35fdb2b146c\/src\libstd\rt.rs:59
  12: panic_unwind::__rust_maybe_catch_panic
             at /rustc/a01e4761a1507939430d4044a5f0a35fdb2b146c\/src\libpanic_unwind\lib.rs:102
  13: std::panicking::try
             at /rustc/a01e4761a1507939430d4044a5f0a35fdb2b146c\/src\libstd\panicking.rs:289
  14: std::panicking::try
             at /rustc/a01e4761a1507939430d4044a5f0a35fdb2b146c\/src\libstd\panicking.rs:289
  15: std::panicking::try
             at /rustc/a01e4761a1507939430d4044a5f0a35fdb2b146c\/src\libstd\panicking.rs:289
  16: main
  17: invoke_main
             at d:\agent\_work\2\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
  18: invoke_main
             at d:\agent\_work\2\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
  19: BaseThreadInitThunk
  20: RtlUserThreadStart
```

Reply:
The backtrace is unrelated. It occurs inside the test framework when tests fail.

I wonder if the attribute we're looking for is removed and simply doesn't exist anymore at this stage in the compiler

Reply:
I think I found the issue. Only macros with `#[macro_export]` on them will actually end up in the HIR. There seems no way around that.

The only other option is to make this entire lint an `EarlyLintPass` (which runs on the AST and thus still knows about all macros).

Sorry, this is not as easy as I thought it would be. If you don't want to continue I understand. If you do want to continue:

1. make this lint an `EarlyLintPass` (and thus also all `LateContext`s to `EarlyContext`s)
2. change `hir::` to `ast::` (only inside this lint. don't touch the other lints in this file)
3. there might be some more parts that don't compile. I don't know what they are in advance, but it should be possible to figure out how to get it working again by looking at the differences between `EarlyLintPass` and `LateLintPass` or going along with what the compiler errors suggest